src/redux/actions/experimentSettings/completeProcessingStep.js

```diff
@@ -1,6 +1,34 @@
 import { EXPERIMENT_SETTINGS_PROCESSING_COMPLETE_STEP } from '../../actionTypes/experimentSettings';
 
-const completeProcessingStep = (experimentId, settingName, numSteps) => (dispatch) => {
+import getApiEndpoint from '../../../utils/apiEndpoint';
+
+const completeProcessingStep = (
+  experimentId,
+  settingName,
+  numSteps,
+) => async (dispatch, getState) => {
+  const { stepsDone } = getState().experimentSettings.processing.meta;
+
+  const arrayStepsDone = Array.from(stepsDone);
+  arrayStepsDone.push(settingName);
+
+  const body = {
+    complete: arrayStepsDone.size === numSteps,
+    stepsDone: arrayStepsDone,
+  };
+
+  await fetch(
+    `${getApiEndpoint()}/v1/experiments/${experimentId}/processingConfig`,
+    {
+      method: 'PUT',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify([{
+        name: 'meta',
+        body,
+      }]),
```

Comment:
In the case it didn't work out should I add a reducer for that and an error message here too? In that case I should probably show an error message when it can't load the steps too.

Reply:
the reducer for the EXPERIMENT_SETTINGS_PROCESSING_ERROR action should already be implemented. In here you will only need to dispatch a `EXPERIMENT_SETTINGS_PROCESSING_ERROR` action. In the component part of data processing, you might need to add handling if the state is errored, but you have to check, maybe it is already implemented, I am not sure. Does that make sense?

Reply:
yes, I seems it is not created

Reply:
yes, `EXPERIMENT_SETTINGS_PROCESSING_ERROR` is only being used to dispatch it, but nothing in reducers takes it right now.

Reply:
yeah, I just took a look and noticed that although there is a reducer for it: `processingSettingsError`, it is not hooked up in `redux/reducers/experimentSettings/index.js`, so you will need to do that too.